File: Service/ClaimCare/RMA.Service.ClaimCare.Contracts/Entities/ClaimTracerInvoice.cs
using RMA.Common.Entities;
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

namespace RMA.Service.ClaimCare.Contracts.Entities
{
    public class ClaimTracerInvoice : AuditDetails
    {
        public int TracerInvoiceId { get; set; }
        public int ClaimId { get; set; }
        public int RolePlayerId { get; set; }
        public decimal? TracingFee { get; set; }
        public int PaymentStatus { get; set; }
        public string Reason { get; set; }
        public System.DateTime? PayDate { get; set; }
        public ClaimInvoiceTypeEnum ClaimInvoiceType { get; set; }
        public System.DateTime? InvoiceDate { get; set; }
        public int PolicyId { get; set; }
        public string PolicyNumber { get; set; }
        public System.DateTime CapturedDate { get; set; }
        public int? ProductId { get; set; }
        public string Product { get; set; }
        public Beneficiary BeneficiaryDetail { get; set; }
        public string MessageText { get; set; }
        public int? BankAccountId { get; set; }
        public string TracerEmail { get; set; }
        public ClaimBankAccountVerification ClaimBankAccountVerification { get; set; }
    }
}

File: Web/RMA-Web-Shell/projects/claimcare/src/app/claim-manager/shared/entities/claimTracerInvoice.ts
import { ClaimInvoiceTypeEnum } from 'projects/shared-models-lib/src/lib/enums/claim-invoice-type-enum';
import { Beneficiary } from 'projects/clientcare/src/app/policy-manager/shared/entities/beneficiary';
import { ClaimBankAccountVerification } from './ClaimBankAccountVerification';

export class ClaimTracerInvoice {
  tracerInvoiceId: number;
  claimId: number;
  rolePlayerId: number;
  tracingFee: number | null;
  paymentStatus: number;
  reason: string;
  payDate: Date;
  claimInvoiceType: ClaimInvoiceTypeEnum;
  invoiceDate: Date;
  policyId: number;
  policyNumber: string;
  capturedDate: Date;
  productId: number | null;
  product: string;
  beneficiaryDetail: Beneficiary;
  messageText: string;
  bankAccountId: number | null;
  tracerEmail: string;
  createdDate: Date;
  claimBankAccountVerification: ClaimBankAccountVerification;
}
